src/app/rpmostree-override-builtins.cxx

```diff
@@ -89,7 +93,58 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
+  g_autoptr(GPtrArray) override_replace_final = g_ptr_array_new();
+  GUnixFDList *list_fds = g_unix_fd_list_new ();
 
+  if (override_replace && opt_ex_pin_from_repos)
+    {
+      g_autoptr(GPtrArray) queries = g_ptr_array_new ();
+
+      for (const char *const* it = override_replace; it && *it; it++)
+        {
+          auto pkg = *it;
+
+          if(rpmostreecxx::is_http_arg (pkg) || rpmostreecxx::is_rpm_arg (pkg))
+            {
+              g_ptr_array_add(override_replace_final, (gpointer)pkg);
+              continue;
+	    }
+          else
+            {
+              g_ptr_array_add(queries, (gpointer)pkg);
+            }
+        }
+      if (queries->len > 0)
+        {
+          g_ptr_array_add (queries, NULL);
+          g_autofree char *refresh_transaction_address = NULL;
+
+          if (!rpmostree_os_call_refresh_md_sync (os_proxy, options, &refresh_transaction_address,
+                                                 cancellable, error))
+            return FALSE;
+          if (!rpmostree_transaction_get_response_sync (sysroot_proxy, (const char*)refresh_transaction_address,
+                                                       cancellable, error))
+            return FALSE;
+
+          if (!rpmostree_osexperimental_call_download_packages_sync (osexperimental_proxy,
+                                                                     (const char *const*) queries->pdata,
+                                                                     NULL, &list_fds,
+                                                                     cancellable, error))
+            return FALSE;
+
+          const int nfds = g_unix_fd_list_get_length (list_fds);
+          GPtrArray *path_list = g_ptr_array_new ();
+
+          for (guint i = 0; i < nfds; i++)
+            {
+              gint fd = g_unix_fd_list_get(list_fds, i, error);
+              g_autofree char *filepath = glnx_fdrel_abspath (fd, ".");
```

Comment:
`glnx_fdrel_abspath` will return an absolute path, but we want something which starts with `file://`. So you can either prepend it by generating a new string, or directly implement what `glnx_fdrel_abspath` does with that tweak.